models/location_helpers.go

```diff
@@ -111,18 +111,93 @@ func FindBackupLocations(q *reform.Querier) ([]*BackupLocation, error) {
 	return locations, nil
 }
 
+// FindBackupLocationByID finds a Backup Location by it's ID.
+func FindBackupLocationByID(q *reform.Querier, id string) (*BackupLocation, error) {
+	if id == "" {
+		return nil, status.Error(codes.InvalidArgument, "Empty Location ID.")
+	}
+
+	location := &BackupLocation{ID: id}
+	switch err := q.Reload(location); err {
+	case nil:
+		return location, nil
+	case reform.ErrNoRows:
+		return nil, status.Errorf(codes.NotFound, "Backup location with ID %q not found.", id)
+	default:
+		return nil, errors.WithStack(err)
+	}
+}
+
+// BackupLocationsConfigs groups all backup locations configs.
+type BackupLocationsConfigs struct {
+	PMMClientConfig *PMMClientLocationConfig
+	PMMServerConfig *PMMServerLocationConfig
+	S3Config        *S3LocationConfig
+}
+
+// Validate checks if there is exactly one config with required fields.
+func (c BackupLocationsConfigs) Validate() error {
+	configCount := 0
```

Comment:
maybe better to check that only one config is set and then validate that config? As a user I can accidentally specify 2 configs and let's say some field in the second config is wrong, so I will try to fix it and only after that I will get an error that I'm setting more than one config.

Reply:
Agree with @oter